### DISCUSSION TITLE
Fastify with Sandpack

### DISCUSSION THREAD

**Discussion by andysm1th** [codesandbox/sandpack#790] [Q&A]
I'm currently trying to run Fastify inside of Sandpack. I've added the required dependencies but get this error:

```
/fastify/server.js: Could not find dependency: 'dns' relative to '/node_modules/fastify/lib/server.js' (1:0)
```

What I don't understand is that Fastify (or the script I'm using) does not require the dns module.

But when I add the required 'dns' package I get the following errors:

```
/fastify/server.js: Error: ENOENT: No such file or directory., '/etc/hosts' (1:0)
/fastify/server.js: Error: ENOENT: No such file or directory., '/etc/resolv.conf' (1:0)
```

Does anyone have any idea why this could be happening or point me in the right direction?

@simoneb

**Comment by danilowoz** [comment]
Hey, thanks for reporting this issue. Could you please provide the minimal files to reproduce this error? 

Thanks

**Comment by andysm1th** [comment]
Hi thanks for the fast response!

For the React component I'm using a default Vite setup with the following:

```tsx
// Vite specific 'raw' import to return file as a string
import server from './fastify/server.js?raw'

<Sandpack 
  customSetup={{ 
    dependencies: { 
      "fastify": "latest"
    },
    entry: "./fastify/server.js"
  }}
  files={{
    './fastify/server.js': server
  }}
/>
```

The Node.js script for Fastify is:

```js
import Fastify from 'fastify'

const start = async function () {
  const fastify = Fastify()

  fastify.get('/', async () => {
    return { hello: 'world' }
  })

  try {
    await fastify.listen({ host: '127.0.0.1', port: 3000 })
  } catch (err) {
    fastify.log.error(err)
    process.exit(1)
  }
}

start()
```

**Reply by danilowoz** [reply]
You need to set a template, otherwise, it will default to the runtime bundler and not Nodebox. For example, the following code works: 

```js
 <Sandpack
      template="node"
      customSetup={{
        dependencies: {
          fastify: "latest"
        },
        entry: "./fastify/server.js"
      }}
      files={{
        "./fastify/server.js": `import Fastify from 'fastify'

const start = async function () {
  const fastify = Fastify()

  fastify.get('/', async () => {
    return { hello: 'world' }
  })

  try {
    await fastify.listen({ host: '127.0.0.1', port: 3000 })
  } catch (err) {
    fastify.log.error(err)
    process.exit(1)
  }
}

start()`
      }}
    />
```

**Reply by andysm1th** [reply]
Perfect this works! Thank-you! I think the terminology threw me a bit here and I thought that a 'template' was just the default script that is returned when using `<Sandbox template="node" />`
